cfme/infrastructure/virtual_machines.py

```diff
@@ -353,40 +353,58 @@ def _nav_to_snapshot_mgmt(self):
         def does_snapshot_exist(self):
             self._nav_to_snapshot_mgmt()
             try:
-                if self.name is not None:
-                    self.snapshot_tree.find_path_to(re.compile(self.name + r".*?"))
-                else:
-                    self.snapshot_tree.find_path_to(re.compile(self.description + r".*?"))
+                self.snapshot_tree.find_path_to(
+                    re.compile(r"{}.*?".format(self.name or self.description)))
                 return True
             except CandidateNotFound:
                 return False
             except NoSuchElementException:
                 return False
 
-        def wait_for_snapshot_active(self):
-            self._nav_to_snapshot_mgmt()
+        def _snapshot_click_helper(self, prop):
+            """Helper method to reduce code duplication.
+
+            Args:
+                prop (str): Property to check (name or description).
+
+            Returns:
+                None
+            """
+            self.snapshot_tree.click_path(
+                *self.snapshot_tree.find_path_to(re.compile(prop)))
+
+        def _snapshot_is_active_helper(self, prop):
+            """Helper for a wait_for_snapshot_active method to reduce code duplication.
+
+            Args:
+                prop (str): Property to check (name or description).
+
+            Returns:
+                bool: True if snapshot is active, False otherwise.
+            """
             try:
-                self.snapshot_tree.click_path(
-                    *self.snapshot_tree.find_path_to(re.compile(self.name)))
-                if sel.is_displayed_text(self.name + " (Active)"):
+                self._snapshot_click_helper(prop)
+                if sel.is_displayed_text("{} (Active)".format(prop)):
                     return True
             except CandidateNotFound:
                 return False
 
+        def wait_for_snapshot_active(self):
```

Comment:
This method doesn't really wait for anything. It just checks if snapshot is active or not, that's it. It should be a `@property` called `active` or `is_active`.